### ISSUE TITLE
Use JSON instead of custom string encoding

### ISSUE THREAD

**Issue by emiltin** [rsmp-nordic/rsmp_core#63]
Several commands pack information into custom strings, instead of using ordinary JSON arrays and hashes.

This can become convoluted, and can be harder to validate with eg. JSON schema

**Comment by emiltin** [comment]
Moving this issue to core. I think the array format should be a general decision that applies to both core and SLXs.

**Comment by emiltin** [comment]
We currently use to types of lists in the tlc sxl: integer lists and boolean lists:
```json
"1,1,4,3",
"True,False,False,True"
```

Using JSON they would look like:
```json
["1", "1", "4", "3"],
["True", "False", "False", "True" ]
```

Or if we use native JSON types for integers and bools:

```json
[1,1,4,3],
[true,false,false,true]
```

**Comment by SwarcoPalm** [comment]
I think the core of the issue is that we're not currently using any form of list type. The field is a string, which is then defined to contain something that can be interpreted as a list. If you define in RSMP core a list or array type, then you can use a normal json serialisation of the list, and you can easily adapt your implementation for other serialisation types in the future.

I suggested something similar earlier as David mentions

**Comment by emiltin** [comment]
`If you define in RSMP core a list or array type`

Do you mean usign build-in JSON types?

**Comment by SwarcoPalm** [comment]
> `If you define in RSMP core a list or array type`
> 
> Do you mean using build-in JSON types?

No, the core specification of RSMP says that for instance status messages can have data of different types.

For instance in the SXL, S0023 List of time plans, is defined as type "string", that means the field will always be transmitted as a string formatted as noted. If we introduce a list type (List of <type>) it can be transferred correctly as a list independant of the serialisation method (JSON / XML/ etc)

**Comment by emiltin** [comment]
Sorry I'm not sure I follow. Can you give an example what the JSON would look like, e.g. when sending S0023?

**Comment by emiltin** [comment]
@SwarcoPalm maybe you mean that we describe in the spec that something should be send e.g as a "list of integers". And then when we serialize as JSON we use native JSON arrays? Then I think we agree.

Spec: "List of Integers"
=>
JSON: [1,2,3,4]


As I see it, it's simply a matter of using native JSON types whenever they are available:

- string
- number
- array (ordered list)
- boolean
- null
- object (unordered key/value pairs)

https://restfulapi.net/json-data-types/
